kotlinx-coroutines-core/jvm/src/internal/LockFreeTaskQueue.kt

```diff
@@ -78,12 +79,17 @@ internal open class LockFreeTaskQueue<E : Any>(
  */
 internal class LockFreeTaskQueueCore<E : Any>(
     private val capacity: Int,
-    private val singleConsumer: Boolean // true when there is only a single consumer (slightly faster)
+    @JvmField // JvmField + public avoids generation of accessors
```

Comment:
We are going to regret maintaining this.
Either R8 or kotlinc should have a flag "escalate visibility to avoid accessors" (dreaming of nestmates)

Reply:
The classes where I did are internal anyway. Yes, it means public in JVM, but it does not make any practical difference, since no public API declares those types in their signature, so you cannot accidentally use them even from Java code (which is highly unlikely to be found around coroutines anyway).

Reply:
My concern is not `public` modifier, but the fact that now we should think about using `@JvmField + public` instead of `private` when the bridge will be generated.
This is fragile, easy to forget and hard to test.

That's why I think some compiler support is required here

Reply:
Agreed. Let's not do this kind of changes and find a more sustainable strategy for this kind of optimizations. One approach is to consider a use of R8 as a standalone optimization tool (it can spit out class-files, too) that we would use on kotlinx-coroutines.jar that we deliver to the public to mangle and shorten the names of all private and internal identifiers and remove accessors. However, we have to be careful that we don't touch anything that might potentially become a part of a user-visible stack-trace. Needs a separate thinking.